### ISSUE TITLE
[2019.09] PHP Fatal Errors

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#7297]
Please report here any PHP Fatal error message obtained running the `2019.06` stable or  `2019.09-dev` versions.

**Comment by AndyHee** [comment]
This is a recent fatal error, details here: https://friendica.hubup.pro/display/174591b4-155d-18c2-9291-98a040514688

```
PHP Notice:  Undefined variable: a in /var/www/friendica/src/Worker/OnePoll.php on line 615
PHP Fatal error:  Uncaught Error: Call to a member function getHostName() on null in /var/www/friendica/src/Worker/OnePoll.php:615
Stack trace:
#0 /var/www/friendica/src/Worker/OnePoll.php(139): Friendica\Worker\OnePoll::pollMail(Array, 1, '2019-06-30 13:1...')
#1 [internal function]: Friendica\Worker\OnePoll::execute(43595)
#2 /var/www/friendica/src/Core/Worker.php(402): call_user_func_array('Friendica\\Worke...', Array)
#3 /var/www/friendica/src/Core/Worker.php(299): Friendica\Core\Worker::execFunction(Array, 'OnePoll', Array, true)
#4 /var/www/friendica/src/Core/Worker.php(114): Friendica\Core\Worker::execute(Array)
#5 /var/www/friendica/bin/worker.php(54): Friendica\Core\Worker::processQueue(false)
#6 {main}
  thrown in /var/www/friendica/src/Worker/OnePoll.php on line 615
```

**Comment by ben-utzer** [comment]
### Expected behavior
Show photo.

### Actual behavior
When I try to open certain photos from /photos/$user/album I get a PHP error.

One block like this shows up in the log each time I press F5 or try to open another photo from the album.

[02-Jul-2019 12:42:31 Europe/Brussels] PHP Fatal error:  Uncaught TypeError: Argument 3 passed to get_responses() must be of the type array, bool given, called in /path/to/webroot/mod/photos.php on line 1541 and defined in https://social.yl.ms/photos/utzer/image/38aea7645d6d44c8b75c558993c99206include/conversation.php:1524
Stack trace:
#0 /path/to/webroot/mod/photos.php(1541): get_responses(Array, Array, false)
#1 /path/to/webroot/src/LegacyModule.php(71): photos_content(Object(Friendica\App))
#2 /path/to/webroot/src/LegacyModule.php(45): Friendica\LegacyModule::runModuleFunction('content')
#3 [internal function]: Friendica\LegacyModule::content()
#4 /path/to/webroot/src/App.php(1241): call_user_func(Array)
#5 /path/to/webroot/index.php(17): Friendica\App->runFrontend()
#6 {main}
  thrown in /path/to/webroot/include/conversation.php on line 1524

### Steps to reproduce the problem
Go to some old photos, in my case it was an album from 2016. 

### Friendica version you encountered the problem
Latest released version, 2019.06.

### Friendica source (git, zip)
git

### PHP version
7.3.6

### SQL version
10.3.16-MariaDB

A copy of https://github.com/friendica/friendica/issues/7340

**Comment by MrPetovan** [comment]
Are you getting a string of Notice messages about undefined indices in `mod/photos.php` as well just before this fatal error?

**Comment by ben-utzer** [comment]
Not that I know of. These would be in the php error log as well?

**Comment by MrPetovan** [comment]
If the `error_reporting` value is set up to log notices as well, yes.

**Comment by ben-utzer** [comment]
Need to check this tomorrow, I never changed the settings what to log in php.ini, but no idea what is set. I will check it tomorrow.

**Comment by ben-utzer** [comment]
No notices, just this as above (??):
`  thrown in /path/to/webroot/include/conversation.php on line 1524
[06-Jul-2019 05:09:05 Europe/Brussels] PHP Fatal error:  Uncaught TypeError: Argument 3 passed to get_responses() must be of the type array, bool given, called in /path/to/webroot/mod/photos.php on line 1541 and defined in /path/to/webroot/include/conversation.php:1524
Stack trace:
#0 /path/to/webroot/mod/photos.php(1541): get_responses(Array, Array, false)
#1 /path/to/webroot/src/LegacyModule.php(71): photos_content(Object(Friendica\App))
#2 /path/to/webroot/src/LegacyModule.php(45): Friendica\LegacyModule::runModuleFunction('content')
#3 [internal function]: Friendica\LegacyModule::content()
#4 /path/to/webroot/src/App.php(1241): call_user_func(Array)
#5 /path/to/webroot/index.php(17): Friendica\App->runFrontend()
#6 {main}
  thrown in /path/to/webroot/include/conversation.php on line 1524`

**Comment by MrPetovan** [comment]
Ok, thanks, the fatal error has been suppressed in https://github.com/friendica/friendica/pull/7342

**Comment by ben-utzer** [comment]
But this does not fix the problem, right?

**Comment by MrPetovan** [comment]
Indeed, we still try to fetch missing item data for old pictures, but at least it doesn't make the page crash.

**Comment by ben-utzer** [comment]
Sorry, but I have to ask again, what do you mean by this?

> at least it doesn't make the page crash

I still get a http 500 error, so this is not what you mean by crash correct?

**Comment by MrPetovan** [comment]
500 error still isn't good, Has it been happening since you recently pulled? Can you provide the (hopefully new) error message triggering the 500 page?

**Comment by ben-utzer** [comment]
Nothing new in the php log, still like this:
`[06-Jul-2019 16:47:59 Europe/Brussels] PHP Fatal error:  Uncaught TypeError: Argument 3 passed to get_responses() must be of the type array, bool given, called in /path/to/webroot/mod/photos.php on line 1541 and defined in /path/to/webroot/include/conversation.php:1524
Stack trace:
#0 /path/to/webroot/mod/photos.php(1541): get_responses(Array, Array, false)
#1 /path/to/webroot/src/LegacyModule.php(71): photos_content(Object(Friendica\App))
#2 /path/to/webroot/src/LegacyModule.php(45): Friendica\LegacyModule::runModuleFunction('content')
#3 [internal function]: Friendica\LegacyModule::content()
#4 /path/to/webroot/src/App.php(1241): call_user_func(Array)
#5 /path/to/webroot/index.php(17): Friendica\App->runFrontend()
#6 {main}
  thrown in /path/to/webroot/include/conversation.php on line 1524
`

Nothing in the httpd log.

**Comment by AlfredSK** [comment]
There are two minor DB updates in develop at the moment. No major changes. Only some bug fixes iirc. The checkout from current master to develop was a no-brainer.

**Comment by ben-utzer** [comment]
> There are two minor DB updates in develop at the moment. No major changes. Only some bug fixes iirc. The checkout from current master to develop was a no-brainer.

I think that will take a long time, the progress in mysql at least is not increasing sind 200 seconds. :-D

**Comment by ben-utzer** [comment]
Update: This shows up in php logs:

`[08-Jul-2019 05:13:07 Europe/Brussels] PHP Notice:  Undefined index: uri in /path/to/webroot/include/conversation.php on line 1529
[08-Jul-2019 05:13:07 Europe/Brussels] PHP Fatal error:  Uncaught BadFunctionCallException: defaults($arr, $key, $def) $key is null in /path/to/webroot/boot.php:348
Stack trace:
#0 /path/to/webroot/include/conversation.php(1529): defaults(Array, NULL, 0)
#1 /path/to/webroot/mod/photos.php(1542): get_responses(Array, Array, Array)
#2 /path/to/webroot/src/LegacyModule.php(71): photos_content(Object(Friendica\App))
#3 /path/to/webroot/src/LegacyModule.php(45): Friendica\LegacyModule::runModuleFunction('content')
#4 [internal function]: Friendica\LegacyModule::content()
#5 /path/to/webroot/src/App.php(1258): call_user_func(Array)
#6 /path/to/webroot/index.php(17): Friendica\App->runFrontend()
#7 {main}
  thrown in /path/to/webroot/boot.php on line 348`

I still get the HTTP 500 fault.

**Comment by MrPetovan** [comment]
Thanks, yes there still is a crash, but it isn't on the same place. Progress!

**Comment by ben-utzer** [comment]
I get a new error and my system just shows the maintenance mode. 

`[09-Jul-2019 06:37:46 Europe/Berlin] PHP Fatal error:  Uncaught Error: Undefined class constant 'Success' in /path/to/webroot/update.php:385
Stack trace:
#0 /path/to/webroot/src/Core/Update.php(187): update_1318()
#1 /path/to/webroot/src/Core/Update.php(139): Friendica\Core\Update::runUpdateFunction(1318, 'update')
#2 /path/to/webroot/src/Core/Update.php(52): Friendica\Core\Update::run('/path/to/webroot')
#3 /path/to/webroot/bin/worker.php(36): Friendica\Core\Update::check('/path/to/webroot', true, Object(Friendica\App\Mode))
#4 {main}
  thrown in /path/to/webroot/update.php on line 385`

**Comment by MrPetovan** [comment]
It shouldn't occur anymore with the latest develop.

**Comment by annando** [comment]
```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Friendica\Util\Network::post() must be of the type string, null given, called in /src/Protocol/DFRN.php on line 1460 and defined in /src/Util/Network.php:253
Stack trace:
#0 /src/Protocol/DFRN.php(1460): Friendica\Util\Network::post(NULL, '{"aes_key":"jhr...', Array)
#1 /src/Worker/Delivery.php(303): Friendica\Protocol\DFRN::transmit(Array, Array, '<?xml version="...', false)
#2 /src/Worker/Delivery.php(181): Friendica\Worker\Delivery::deliverDFRN('wall-new', Array, Array, Array, Array, true, false, false)
#3 [internal function]: Friendica\Worker\Delivery::execute('wall-new', 32326353, 845808)
#4 /src/Core/Worker.php(402): call_user_func_array('Friendica\\Worke...', Array)
#5 /src/Core/Worker.php(299): Frien in /src/Util/Network.php on line 253

```

**Comment by annando** [comment]
```
[15-Jul-2019 20:13:30 Europe/Berlin] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Friendica\Core\Config\Cache\PConfigCache::get() must be of the type integer, null given, called in /src/Core/Config/PConfiguration.php on line 87 and defined in /src/Core/Config/Cache/PConfigCache.php:65
Stack trace:
#0 /src/Core/Config/PConfiguration.php(87): Friendica\Core\Config\Cache\PConfigCache->get(NULL, 'feature', 'categories')
#1 /src/Core/PConfig.php(62): Friendica\Core\Config\PConfiguration->get(NULL, 'feature', 'categories', false, false)
#2 /src/Content/Feature.php(28): Friendica\Core\PConfig::get(NULL, 'feature', 'categories', false)
#3 /src/Content/Widget.php(300): Friendica\Content\Feature::isEnabled(NULL, 'categories')
#4 /src/Module/Profile.php(198): Friendica\Content\Widget::categories('https://pirati....', '')
#5 [internal in /src/Core/Config/Cache/PConfigCache.php on line 65
```

**Comment by JeroenED** [comment]
```
PHP Fatal error:  Uncaught TypeError: Argument 3 passed to Friendica\\Core\\L10n::tt() must be of the type int, null given, called in /home/jeroen/domains/jeroened.be/social/web/src/Content/Widget.php on line 58 and defined in /home/jeroen/domains/jeroened.be/social/web/src/Core/L10n.php:106\nStack trace:
#0 /home/jeroen/domains/jeroened.be/social/web/src/Content/Widget.php(58): Friendica\\Core\\L10n::tt('%d invitation a...', '%d invitations ...', NULL)
#1 /home/jeroen/domains/jeroened.be/social/web/src/Module/Contact.php(349): Friendica\\Content\\Widget::findPeople()
#2 [internal function]: Friendica\\Module\\Contact::content()
#3 /home/jeroen/domains/jeroened.be/social/web/src/App.php(1269): call_user_func(Array)
#4 /home/jeroen/domains/jeroened.be/social/web/index.php(17): Friendica\\App->runFrontend()
#5 {main}
 thrown in /home/jeroen/domains/jeroened.be/social/web/src/Core/L10n.php on line 106
```

**Comment by AlfredSK** [comment]
Currently, the image upload is broken in develop:

```
[29-Jul-2019 19:08:10 Europe/Berlin] PHP Fatal error:  Uncaught Error: Call to undefined method Friendica\Model\Photo::select() in .../mod/photos.php:638
Stack trace:
#0 .../src/LegacyModule.php(71): photos_post(Object(Friendica\App))
#1 .../src/LegacyModule.php(50): Friendica\LegacyModule::runModuleFunction('post')
#2 [internal function]: Friendica\LegacyModule::post()
#3 .../src/App.php(1220): call_user_func(Array)
#4 ...index.php(20): Friendica\App->runFrontend()
#5 {main}
  thrown in .../mod/photos.php on line 638
```

**Comment by MrPetovan** [comment]
Paging @annando again.

**Comment by copiis** [comment]
Just get this on master

```
PHP Fatal error:  Uncaught Error: Class 'Dice\Dice' not found in /bin/worker.php:34
Stack trace: #0 {main} thrown in /bin/worker.php on line 34
```

**Comment by AlfredSK** [comment]
That shouldn't be possible because DICE was introduced in develop lately. Are you sure you are on master and your add-ons too?

**Comment by copiis** [comment]
I tryed to switch to develop but i got this
```
[RuntimeException]                                                            
  Failed to execute git clone --mirror 'git+ssh://git@github.com/stanlemon/jGr  
  owl.git' '/home/users/meld/.cache/composer/vcs/git-ssh---git-github.com-stan  
  lemon-jGrowl.git/'                                                            
                                                                                
  Cloning into bare repository '/home/users/meld/.cache/composer/vcs/git-ssh--  
  -git-github.com-stanlemon-jGrowl.git'...                                      
  git@github.com: Permission denied (publickey).                                
  fatal: Could not read from remote repository.                                 
                                                                                
  Please make sure you have the correct access rights                           
  and the repository exists. 
```
Afther that iw switched back to master ant after that i get that error message

**Comment by MrPetovan** [comment]
Please check this GitHub page for possible troubleshooting: https://help.github.com/en/articles/error-permission-denied-publickey

**Comment by copiis** [comment]
Latest develop

```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Friendica\Model\Contact::unmarkForArchival() must be of the type array, boolean given, called in a/src/Model/Contact.php on line 1818 and defined in /src/Model/Contact.php:879
Stack trace:
#0 /src/Model/Contact.php(1818): Friendica\Model\Contact::unmarkForArchival(false)
#1 /src/Model/Contact.php(2017): Friendica\Model\Contact::updateContact(226787, 0, 'https://fedi.po...', Array)
#2 /src/Worker/UpdateContact.php(20): Friendica\Model\Contact::updateFromProbe(226787, '', true)
#3 [internal function]: Friendica\Worker\UpdateContact::execute(226787, 'force')
#4 /src/Core/Worker.php(399): call_user_func_array('Friendica\\Worke...', Array)
#5/src/Core/Worker.php(298): Friendica\Core\Worker::execFunction(Array, 'UpdateContact', Array, true)
#6 /src/Model/Contact.php on line 879
```

**Comment by MrPetovan** [comment]
On it.

**Comment by MrPetovan** [comment]
If it doesn't happen that often, it may just be a dropped database connection. Nothing we can do, the fatal error actually is warranted.

**Comment by AlfredSK** [comment]
develop:

```
Uncaught ImagickException: no decode delegate for this image format `' @ error/constitute.c/ReadImage/504 in .../src/Object/Image.php:750
Stack trace:
#0 .../src/Object/Image.php(750): Imagick->__construct('photo')
#1 .../src/Content/Text/BBCode.php(440): Friendica\Object\Image::guessType('photo', true)
#2 .../src/Content/Text/Markdown.php(122): Friendica\Content\Text\BBCode::scaleExternalImages('[h3]Le canal d'...', false)
#3 .../src/Protocol/Diaspora.php(2922): Friendica\Content\Text\Markdown::toBBCode('[h3]Le canal d'...')
#4 .../src/Protocol/Diaspora.php(757): Friendica\Protocol\Diaspora::receiveStatusMessage(Array, Object(SimpleXMLElement), '<status_message...')
#5 .../src/Protocol/Diaspora.php(655): Friendica\Protocol\Diaspora::dispatch(Array, Array, Object(SimpleXMLElement))
#6 .../src/Protocol/Diaspora.php(1354): Friendica\Protocol\Diaspora::dispatchPu in .../src/Object/Image.php on line 750
```

**Comment by MrPetovan** [comment]
Seems specifically relevant to the post in question, would you be able to pull its source from the little the exception shows if it's a public post?

**Comment by AlfredSK** [comment]
Where do I have to look for it? In 'item-content' -> 'body' or somewhere else?

**Comment by MrPetovan** [comment]
Yes, then you can retrieve the GIUD and paste it in `/admin/item/source`

**Comment by AlfredSK** [comment]
Hmm. Empty result set. Does the body field content start with '[h3]Le canal'?
When I search for '[h3]Le ca' I get results but not for '[h3]Le canal'.

**Comment by MrPetovan** [comment]
What about searching for `%Le canal d'%`?

**Comment by AlfredSK** [comment]
Nothing found for the issue above. But I have a new one. :-)

```
Uncaught Error: Class 'Friendica\Module\Search\Login' not found in .../src/Module/Search/Directory.php:18
Stack trace:
#0 [internal function]: Friendica\Module\Search\Directory::content()
#1 .../src/App/Page.php(311): call_user_func(Array)
#2 .../src/App/Page.php(365): Friendica\App\Page->initContent(Object(Friendica\App\Module), Object(Friendica\App\Mode))
#3 .../src/App.php(742): Friendica\App\Page->run(Object(Friendica\App), Object(Friendica\App\BaseURL), Object(Friendica\App\Mode), Object(Friendica\App\Module), Object(Friendica\Core\L10n\L10n), Object(Friendica\Core\Config\PreloadConfiguration), Object(Friendica\Core\Config\PreloadPConfiguration))
#4 .../index.php(24): Friendica\App->runFrontend(Object(Friendica\App\Module), Object(Friendica\App\Router), Object(Friendica\Core\Config\PreloadPConfiguration))
#5 {main}
  thrown in .../src/Module/Search/Directory.php on line 18
```

**Comment by MrPetovan** [comment]
On it.

**Comment by copiis** [comment]
```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Friendica\Model\Contact::unmarkForArchival() must be of the type array, boolean given, called in /src/Model/Contact.php on line 1818 and defined in a/src/Model/Contact.php:879
Stack trace:
#0 /src/Model/Contact.php(1818): Friendica\Model\Contact::unmarkForArchival(false)
#1 /src/Model/Contact.php(2017): Friendica\Model\Contact::updateContact(256476, 0, 'https://diasp.o...', Array)
#2 /src/Worker/UpdateContact.php(20): Friendica\Model\Contact::updateFromProbe(256476, '', true)
#3 [internal function]: Friendica\Worker\UpdateContact::execute(256476, 'force')
#4 /src/Core/Worker.php(400): call_user_func_array('Friendica\\Worke...', Array)
#5 /src/Core/Worker.php(299): Friendica\Core\Worker::execFunction(Array, 'UpdateContact', Array, true)
#6 /home/users/meld/www/frie in /home/users/meld/www/friendica/src/Model/Contact.php on line 879
```

**Comment by ben-utzer** [comment]
I got a problem that the maximum number of processes that can be run was reached, not sure if this fault might be the cause, but maybe someone understands the error. 

`[19-Aug-2019 20:10:25 Europe/Berlin] PHP Fatal error:  Uncaught Error: Call to a member function prepare() on null in /web/path/src/Database/Database.php:503
Stack trace:
#0 /web/path/src/Database/Database.php(630): Friendica\Database\Database->p('SELECT `jobs` F...', Array)
#1 /web/path/src/Database/Database.php(1453): Friendica\Database\Database->p('SELECT `jobs` F...', Array)
#2 /web/path/src/Database/Database.php(1382): Friendica\Database\Database->select('worker-ipc', Array, Array, Array)
#3 /web/path/src/Database/DBA.php(399): Friendica\Database\Database->selectFirst('worker-ipc', Array, Array, Array)
#4 /web/path/src/Core/Worker.php(1321): Friendica\Database\DBA::selectFirst('worker-ipc', Array, Array)
#5 /web/path/bin/daemon.php(193): Friendica\Core\Worker::IPCJobsExists()
#6 {main}
  thrown in /web/path/src/Database/Database.php on line 503`

**Comment by AlfredSK** [comment]
RC:

```
PHP Fatal error:  Uncaught Error: Call to undefined method Friendica\Core\L10n::getPokeVerbs() in .../mod/poke.php:165
Stack trace:
#0 .../src/LegacyModule.php(71): poke_content(Object(Friendica\App))
#1 .../src/LegacyModule.php(45): Friendica\LegacyModule::runModuleFunction('content')
#2 [internal function]: Friendica\LegacyModule::content()
#3 .../src/App/Page.php(311): call_user_func(Array)
#4 .../src/App/Page.php(365): Friendica\App\Page->initContent(Object(Friendica\App\Module), Object(Friendica\App\Mode))
#5 .../src/App.php(742): Friendica\App\Page->run(Object(Friendica\App), Object(Friendica\App\BaseURL), Object(Friendica\App\Mode), Object(Friendica\App\Module), Object(Friendica\Core\L10n\L10n), Object(Friendica\Core\Config\PreloadConfiguration), Object(Friendica\Core\Config\PreloadPConfiguration))
#6 .../index.php(24): Friendica\App->runFrontend(Object(Friendica\App\Module), Object(Friend in .../mod/poke.php on line 165
```

**Comment by AlfredSK** [comment]
RC:

```
PHP Fatal error:  Uncaught Error: Class 'Friendica\Module\Register' not found in .../config/local-sample.config.php:37
Stack trace:
#0 {main}
  thrown in .../config/local-sample.config.php on line 37
```

**Comment by MrPetovan** [comment]
No one should be accessing this file, let alone directly from a browser like this, you can ignore it.

**Comment by nupplaphil** [comment]
Closed in favor of #7675
